R/redist_map.R

```diff
@@ -0,0 +1,396 @@
+##############################################
+## Author: Cory McCartan
+## Institution: Harvard University
+## Date Created: 2021/01/28
+## Purpose: redist functions for a tidy workflow
+##############################################
+
+#######################
+# constructors and reconstructors
+
+# Main internal constructor
+new_redist_map = function(data, adj, ndists, pop_bounds, pop_col="pop",
+                          adj_col="adj", add_adj=TRUE, existing_col=NULL) {
+    if (add_adj) {
+        stopifnot(!is.null(adj))
+
+        data[[adj_col]] = adj
+    }
+
+    stopifnot(is.integer(ndists))
+    stopifnot(is.numeric(pop_bounds))
+    stopifnot(length(pop_bounds) == 3)
+
+    data = reconstruct.redist_map(data)
+    attr(data, "ndists") = ndists
+    attr(data, "pop_bounds") = pop_bounds
+    attr(data, "pop_col") = pop_col
+    attr(data, "adj_col") = adj_col
+    attr(data, "existing_col") = existing_col
```

Comment:
Did this need to be updated? It may not be, but I'm not sure where this ended up.